keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/verification/ContainerTestVerificationEvaluatorTests.kt

```diff
@@ -0,0 +1,164 @@
+package com.netflix.spinnaker.keel.titus.verification
+
+import com.netflix.spinnaker.keel.api.actuation.Task
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.titus.TitusServerGroup.Location
+import com.netflix.spinnaker.keel.api.verification.VerificationContext
+import com.netflix.spinnaker.keel.api.verification.VerificationState
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.FAILED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.PASSED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.RUNNING
+import com.netflix.spinnaker.keel.orca.ExecutionDetailResponse
+import com.netflix.spinnaker.keel.orca.OrcaExecutionStatus
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.test.deliveryConfig
+import com.netflix.spinnaker.keel.titus.batch.RUN_JOB_TYPE
+import de.huxhorn.sulky.ulid.ULID
+import io.mockk.mockk
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import strikt.api.expectCatching
+import strikt.api.expectThat
+import strikt.assertions.get
+import strikt.assertions.isEqualTo
+import strikt.assertions.isSuccess
+import java.time.Instant.now
+import io.mockk.coEvery as every
+import io.mockk.coVerify as verify
+
+internal class ContainerTestVerificationEvaluatorTests {
+
+  private val orca = mockk<OrcaService>()
+  private val taskLauncher = mockk<TaskLauncher>()
+  private val subject = ContainerTestVerificationEvaluator(orca, taskLauncher)
+
+  private val context = VerificationContext(
+    deliveryConfig = deliveryConfig(),
+    environmentName = "test",
+    artifactReference = "fnord",
+    version = "1.1"
+  )
+  private val verification = ContainerTestVerification(
+    repository = "illuminati/fnord",
+    location = Location(
+      account = "titustestvpc",
+      region = "ap-south-1"
+    )
+  )
+
+  @Test
+  fun `starting verification launches a container job via orca`() {
+    val taskId = ULID().nextULID()
+
+    every {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = any()
+      )
+    } answers { Task(id = taskId, name = arg(3)) }
+
+    expectCatching {subject.start(context, verification)}
+      .isSuccess()
+      .get("taskId") isEqualTo taskId
+
+    verify {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = match {
+          it.first()["type"] == RUN_JOB_TYPE
```

Comment:
It'd be nice to check that at least the container image matches what was specified. 🙂

Reply:
Is it really testing any logic or just that we can copy fields from one place to another?

Reply:
Not really, but you could say the same about the check for the stage type you have there. I just think it helps the test tell the story, is all. Not critical.